src/config.ts

```diff
@@ -342,29 +344,61 @@ const EXCLUDED_FEE_SOURCES = (() => {
     }
 })();
 
+const EXCHANGE_PROXY_OVERHEAD_NO_VIP = () => new BigNumber(150e3);
+const EXCHANGE_PROXY_OVERHEAD_NO_MULTIPLEX = (sourceFlags: number) => {
+    if ([SOURCE_FLAGS.Uniswap_V2, SOURCE_FLAGS.SushiSwap].includes(sourceFlags)) {
+        return TX_BASE_GAS;
+    } else if (SOURCE_FLAGS.LiquidityProvider === sourceFlags) {
+        return TX_BASE_GAS.plus(10e3);
+    } else {
+        return new BigNumber(150e3);
+    }
+};
+const MULTIPLEX_BATCH_FILL_SOURCE_FLAGS =
+    SOURCE_FLAGS.Uniswap_V2 | SOURCE_FLAGS.SushiSwap | SOURCE_FLAGS.LiquidityProvider | SOURCE_FLAGS.RfqOrder;
+const MULTIPLEX_MULTIHOP_FILL_SOURCE_FLAGS =
+    SOURCE_FLAGS.Uniswap_V2 | SOURCE_FLAGS.SushiSwap | SOURCE_FLAGS.LiquidityProvider;
+const EXCHANGE_PROXY_OVERHEAD_FULLY_FEATURED = (sourceFlags: number) => {
+    if ([SOURCE_FLAGS.Uniswap_V2, SOURCE_FLAGS.SushiSwap].includes(sourceFlags)) {
+        // Uniswap VIP
+        return TX_BASE_GAS;
+    } else if (SOURCE_FLAGS.LiquidityProvider === sourceFlags) {
+        // PLP VIP
+        return TX_BASE_GAS.plus(10e3);
+    } else if ((MULTIPLEX_BATCH_FILL_SOURCE_FLAGS | sourceFlags) === MULTIPLEX_BATCH_FILL_SOURCE_FLAGS) {
+        // Multiplex batch fill
+        return TX_BASE_GAS.plus(20e3);
+    } else if (
+        (MULTIPLEX_MULTIHOP_FILL_SOURCE_FLAGS | sourceFlags) ===
+        (MULTIPLEX_MULTIHOP_FILL_SOURCE_FLAGS | SOURCE_FLAGS.MultiHop)
+    ) {
+        // Multiplex multi-hop fill
+        return TX_BASE_GAS.plus(20e3);
+    } else {
+        return new BigNumber(150e3);
+    }
+};
+
 export const ASSET_SWAPPER_MARKET_ORDERS_OPTS: Partial<SwapQuoteRequestOpts> = {
```

Comment:
Just curious, isnt' it cleaner to have a `BASE_ ASSET_SWAPPER_MARKET_ORDERS_OPTS` and then have concrete types rhat extend from this?